### PR TITLE
Feat/orders page

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- Create getOrders page UI/UX. Connect Fe to Be. apply sorting, filtering and pagination features

## Issue ticket code (and/or) and link

- [UI task](https://codecraftersintership.atlassian.net/browse/SCRUM-37)
- [connect Fe to Be task](https://codecraftersintership.atlassian.net/browse/SCRUM-39)

## Example
orders-page

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/Orders/api/getOrders.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}: {
+  sortBy: string;
+  filter: string;
+  page: number;
+  companyId: number;
+}): Promise<void> => {
+  try {
+    store.dispatch(setisVisible(true));
+    const orders: AxiosResponse = await axios.get(
+      `http://localhost:4000/orders/?sortBy=${sortBy}${filter !== 'STATUS' ? `&filterBy=${filter}` : ''}&page=${page}&companyId=${companyId}`,
```
Comment (lyudad) [inline src/pages/Orders/api/getOrders.ts]: 1.1.1

File: src/pages/Orders/components/OrderItem/OrderItem.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  customerBlock,
+  collectionDateStyles,
+  collectionAddressStyles,
+  routeBlock,
+  luggageStyles,
+  statusBlock,
+  collectionTimeStyles,
+  popup,
+} from '@/pages/Orders/components/OrderItem/styles';
+import { FC } from 'react';
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/OrderItem.tsx]: react always first

File: src/pages/Orders/components/OrderItem/OrderItem.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  collectionTimeEnd,
+  collectionAddress,
+  customer,
+  status,
+  luggages,
+  routeId,
+}) => {
+  const { t } = useTranslation();
+  const customerNames: string[] = customer.full_name.split(' ');
+  const routeIdView = !routeId ? 'NEW' : routeId.id;
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/OrderItem.tsx]: 1.3.3

File: src/pages/Orders/components/OrderItem/OrderItem.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  const timeStart = new Date(collectionTimeStart);
+  const timeEnd = new Date(collectionTimeEnd);
+
+  return (
+    <Box sx={orderRow(!routeId)}>
+      <Typography sx={collectionDateStyles}>
+        {date.getDate()} {MONTHS[date.getMonth()]} {date.getFullYear()}
+      </Typography>
+      <Typography sx={collectionTimeStyles}>
+        {timeStart.getHours()}:00 - {timeEnd.getHours()}:00
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/OrderItem.tsx]: I dont think its right set 00, what if its 30 min

File: src/pages/Orders/components/OrderItem/styles.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { SxProps, Theme } from '@mui/material/styles';
+
+export const orderRow: (isNew: boolean) => SxProps<Theme> = (isNew) => {
+  return {
+    width: '1236px',
+    display: 'flex',
+    marginBottom: '-1px',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    background: isNew ? COLORS.lightPurple : '#FFF',
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/styles.ts]: 1.3.2

File: src/pages/Orders/components/OrderItem/styles.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  '&:hover': {
+    '& .popup': {
+      display: 'block',
+    },
+  },
+};
+
+export const popup: SxProps<Theme> = {
+  backgroundColor: '#fff',
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/styles.ts]: 1.3.2

File: src/pages/Orders/components/OrderItem/styles.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  position: 'absolute',
+  textAlign: 'center',
+  top: '-60px',
+  padding: '8px',
+  zIndex: 9,
+  border: '1px solid #ccc',
```
Comment (lyudad) [inline src/pages/Orders/components/OrderItem/styles.ts]: 1.3.2

File: src/pages/Orders/components/SortingRow/useSortOrders.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  setParamsFilter,
+  setParamsSortBy,
+} from '@/store/slices/ordersPageSlice';
+import { ORDERS_SORTS } from '@/constants/ordersSorts';
+import { getOrders } from '../../api/getOrders';
```
Comment (lyudad) [inline src/pages/Orders/components/SortingRow/useSortOrders.ts]: 1.3.5

File: src/pages/Orders/components/SortingRow/useSortOrders.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+import { RootState } from '@/store/store';
```
Comment (lyudad) [inline src/pages/Orders/components/SortingRow/useSortOrders.ts:9]: 1.1.14

File: src/pages/Orders/index.tsx [hunk 10]
```diff
@@ -1,12 +1,41 @@
 import React from 'react';
-import { useTranslation } from 'react-i18next';
+import OrderItem from '@/pages/Orders/components/OrderItem/OrderItem';
+import SortingRow from '@/pages/Orders/components/SortingRow/SortingRow';
+import { Box, Pagination } from '@mui/material';
+import { useOrdersPagination } from './useOrdersPagination';
 
 function OrdersPage(): React.ReactElement {
-  const { t } = useTranslation();
+  const { viewData, currentPage, totalPages, fetchOrders } =
```
Comment (lyudad) [inline src/pages/Orders/index.tsx]: rename viewData to something more self-explainable,like: oders, odersList, ordersDetails

File: src/components/Loader/styles.css [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+.loader:before,
+.loader:after {
+  border-radius: 50%;
+  width: 2.5em;
+  height: 2.5em;
+  animation-fill-mode: both;
+  animation: bblFadInOut 1.8s infinite ease-in-out;
+}
+.loader {
+  color: #fff;
```
Comment (lyudad) [inline src/components/Loader/styles.css]: 1.3.2

File: src/pages/Orders/components/SearchOrders/index.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <Box sx={formWrapper}>
+      <FormProvider {...methods}>
+        <img
+          style={{ cursor: 'pointer' }}
+          src={searchIcon}
+          alt="searchIcon"
+          onClick={methods.handleSubmit((data) => {
+            console.log(data);
```
Comment (lyudad) [inline src/pages/Orders/components/SearchOrders/index.tsx:21]: delete

File: src/pages/Orders/components/SortingRow/useSortOrders.ts (hunk 9 above)
Comment (lyudad) [reply src/pages/Orders/components/SortingRow/useSortOrders.ts:9]: I dont see any comment to my comment

File: src/pages/Orders/components/SearchOrders/index.tsx (hunk 12 above)
Comment (Parle648) [reply src/pages/Orders/components/SearchOrders/index.tsx:21]: done

File: src/components/Loader/styles.css (hunk 11 above)
Comment (Parle648) [reply src/components/Loader/styles.css]: fix

File: src/pages/Orders/index.tsx (hunk 10 above)
Comment (Parle648) [reply src/pages/Orders/index.tsx]: renamed

File: src/pages/Orders/components/SortingRow/useSortOrders.ts (hunk 9 above)
Comment (Parle648) [reply src/pages/Orders/components/SortingRow/useSortOrders.ts:9]: done

File: src/pages/Orders/components/SortingRow/useSortOrders.ts (hunk 8 above)
Comment (Parle648) [reply src/pages/Orders/components/SortingRow/useSortOrders.ts]: done

File: src/pages/Orders/components/OrderItem/styles.ts (hunk 7 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/styles.ts]: done

File: src/pages/Orders/components/OrderItem/styles.ts (hunk 6 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/styles.ts]: done

File: src/pages/Orders/components/OrderItem/styles.ts (hunk 5 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/styles.ts]: done

File: src/pages/Orders/components/OrderItem/OrderItem.tsx (hunk 4 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/OrderItem.tsx]: done

File: src/pages/Orders/components/OrderItem/OrderItem.tsx (hunk 3 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/OrderItem.tsx]: done

File: src/pages/Orders/components/OrderItem/OrderItem.tsx (hunk 2 above)
Comment (Parle648) [reply src/pages/Orders/components/OrderItem/OrderItem.tsx]: done

File: src/pages/Orders/api/getOrders.ts (hunk 1 above)
Comment (Parle648) [reply src/pages/Orders/api/getOrders.ts]: done

**Pull request merged**